python/lsst/daf/persistence/registries.py

```diff
@@ -363,6 +368,37 @@ def lookup(self, lookupProperties, reference, dataId, **kwargs):
         cursor.execute(cmd, valueList)
         return [row for row in cursor.fetchall()]
 
+    def lookup(self, lookupProperties, reference, dataId, **kwargs):
+        """Perform a lookup in the registry.
+
+        Return values are refined by the values in dataId.
+        Returns a list of values that match keys in lookupProperties.
+        e.g. if the template is 'raw/raw_v%(visit)d_f%(filter)s.fits.gz', and
+        dataId={'visit':1}, and lookupProperties is ['filter'], and the
+        filesystem under self.root has exactly one file 'raw/raw_v1_fg.fits.gz'
+        then the return value will be [('g',)]
+
+        :param lookupProperties:
+        :param dataId: must be an iterable. Keys must be string.
+        If key is a string then will look for elements in the repository that match value for key.
+        If key is a 2-item iterable then will look for elements in the repository where the value is between
+        the values of key[0] and key[1].
```

Comment:
```suggestion
        :param dataId: must be a key/value iterable. Keys must be string.
        If value is a string then will look for elements in the repository that match that value.
        If value is a 2-item iterable then will look for elements in the repository where the value is between
        the values of value[0] and value[1].
```

Reply:
Kind of duplicative to have this both here and above; maybe remove one?

Reply:
Done